File: app/src/test/java/com/khmelenko/lab/varis/log/TextLeafTest.java
package com.khmelenko.lab.varis.log;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class TextLeafTest {
    @Test
    public void addAnsiCodes() throws Exception {
        TextLeaf actual = new TextLeaf("Magenta Italic", FormattingOptions.fromAnsiCodes(new String[]{"35", "3"}));
        TextLeaf expected = new TextLeaf("Magenta Italic");
        expected.getOptions().setTextColor("#FF73FD");
        expected.getOptions().setItalic(true);
        assertEquals(expected.toString(), actual.toString());
    }

    @Test
    public void toHtml() throws Exception {
        TextLeaf text = new TextLeaf("T", FormattingOptions.fromAnsiCodes(new String[]{"34", "1"}));
        assertEquals("<b><font color='#96CBFE'>T</font></b>", text.toHtml());
    }

}

File: app-v3/src/test/java/com/khmelenko/lab/varis/log/TextLeafTest.kt
package com.khmelenko.lab.varis.log

import org.junit.Test

import org.junit.Assert.assertEquals

class TextLeafTest {
    @Test
    fun addAnsiCodes() {
        val actual = TextLeaf("Magenta Italic", FormattingOptions.fromAnsiCodes(*arrayOf("35", "3")))
        val expected = TextLeaf("Magenta Italic")
        expected.options.textColor = "#FF73FD"
        expected.options.isItalic = true
        assertEquals(expected.toString(), actual.toString())
    }

    @Test
    fun toHtml() {
        val text = TextLeaf("T", FormattingOptions.fromAnsiCodes(*arrayOf("34", "1")))
        assertEquals("<b><font color='#96CBFE'>T</font></b>", text.toHtml())
    }
}
